lake/oneshot_cls.py

```diff
@@ -168,16 +187,31 @@ def main():
   # ---------------------------------------------------------------------------
   print('-------- Few-shot Evaluation (Study and Recall) ---------')
 
-  # Prepare data loaders
-  study_loader = torch.utils.data.DataLoader(
-    OmniglotOneShotDataset('./data', train=True, download=True,
-                           transform=image_tfms, target_transform=None),
-    batch_size=config['study_batch_size'], shuffle=False)
-
-  recall_loader = torch.utils.data.DataLoader(
-    OmniglotOneShotDataset('./data', train=False, download=True,
-                           transform=image_tfms, target_transform=None),
-    batch_size=config['study_batch_size'], shuffle=False)
+  if dataset_name == 'Omniglot':
```

Comment:
It actually isn't more concise to do this because we still have to load two datasets per condition (one for study and one for recall), and because each data loader takes different arguments (CIFAR data loader needs classes for our tests and also has an extra mode for validation) we still need to instantiate 2 datasets per condition. So if we change it to instantiate the data loader in one place, it ends up being more lines of code. 

Having said this, it may be more clear what is happening if the data loader is instantiated outside of the 'if' conditions. So I think it is still worth making this change, not for conciseness but for clarity.